src/commonMain/kotlin/com/github/nanoflakes/NanoflakeGenerator.kt

```diff
@@ -5,25 +5,19 @@ package com.github.nanoflakes
  */
 interface NanoflakeGenerator {
     /**
-     * Gets the next [nanoflake][Nanoflake].
-     *
-     * @return a new, generated nanoflake.
+     * This nanoflake generator's epoch.
      */
-    operator fun next(): Nanoflake
+    val epoch: Long
 
     /**
-     * Gets this nanoflake generator's epoch.
-     *
-     * @return the epoch in milliseconds.
+     * This nanoflake generator's epoch as a [DateTime].
      */
-    fun epochMillis(): Long
+    val epochDate: DateTime get() = epoch.toDateTime()
 
     /**
-     * Gets this nanoflake generator's epoch as an [OffsetDateTime].
+     * Gets the next [nanoflake][Nanoflake].
      *
-     * @return a [OffsetDateTime] set to the epoch time.
+     * @return a new, generated nanoflake.
      */
-    fun NanoflakeGenerator.epoch(): DateTime {
```

Comment:
I believe this was some kind of error during the Java -> Kotlin porting process. This method doesn't make a lot of sense. It can only be used when the caller is in a scope where the receiver is of type NanoflakeGenerator. So basically something like this

```kotlin
val generator = Nanoflakes.localGenerator(0, 0)
with(generator) {
    epoch()
    // or
    generator.epoch()
}
```
They would not work without being in that `with` block.

Reply:
Yeah, this is definitively an error on my part